packages/std/src/ibc.rs

```diff
@@ -215,11 +213,21 @@ impl IbcAcknowledgement {
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
 pub struct IbcChannelOpenMsg {
     pub channel: IbcChannel,
+    /// CounterpartyVersion can be None when not known this context, yet
+    pub counterparty_version: Option<String>,
 }
 
 impl IbcChannelOpenMsg {
     pub fn new(channel: IbcChannel) -> Self {
-        Self { channel }
+        Self {
+            channel,
+            counterparty_version: None,
+        }
+    }
+
+    pub fn counterparty_version(mut self, counterparty_version: impl Into<String>) -> Self {
```

Comment:
Hmm, isn't `.field_name()` typically used as a getter to the field?

In `SubMsg`/`MockQuerier` we have `with_gas_limit`/`with_custom_handler`, i.e. `.with_field_name(` as the name of the builder method.

Reply:
I guess it's used as a setter in the builder pattern where you'd have a separate `IbcChannelBuilder` type with all the setters and then use a `finalize` method to build an `IbcChannel` out of that. There's no confusion then. Hm.

Reply:
Yeah, I like the `with_`.

Alternatively, for some things we could at some point consider the proper builder pattern. See https://docs.rs/derive_builder/0.10.2/derive_builder/